polar_route/crossing_smoothing.py

```diff
@@ -0,0 +1,1437 @@
+            if len(_connections) == 1 and (abs(add_case_a) == abs(add_case_b)):
+               u_connections = [cell_a_neighbour,list(_connections)[0]]
+               return list(u_connections),[add_case_a,case,-add_case_b]
+
+        return None,None
+                
+
+    def _neighbour_cells(self,cell_a,cell_b,case,add_case_a,add_case_b):
+        '''
+            Adding in the neighbour cell information as a dict and case types of the neighbour cells that must be 
+            added. If the add_indicies is None then this means that the case need to change relating the adjacency
+            cell pair, but no additional cells need to be added
+
+            Input:
+                cell_a (dict) - Start cell environmental info as dictionary
+                cell_b (dict) - End cell environmental info as dictionary
+                case (int) - Adjaceny case tupe connecting the two cells
+                add_case_a (int) - Possible additional case edge relative to start cell to add
+                add_case_b (int) - Possible additional case edge relative to start cell to add
+            
+            Returns
+                additional_indices (list) - A list of possible cell dictionary info. None if no index added.
+                additional_cases (list) - A list of the cases connecting the additional cell indices. None if no index added.
+        '''
+
+
+        add_indices,add_cases = self._neighbour_indices(cell_a,cell_b,case,add_case_a,add_case_b)
+
+        if add_indices == None:
+            return None,add_cases
+        
+        else:
+            return [self.dijkstra_graph[ii] for ii in add_indices],add_cases
+
+    def nearest_neighbour(self,start,end,case,x):
+        '''
+            Returns the cell in the mesh that shares a boundary with cellA and has an edge on the line that extends the common 
+            boundary of cellA and cellB (and on which the point x lies) in the direction of x. 
+            If x lies inside cellA or there is no cell that satisfies these requirements, it returns null.
+
+            Input:
+                start (dict) - Start cell environmental info as dictionary
+                end (dict)   - End cell environmental info as dictionary
+                case (int)   - Adjaceny case tupe connecting the two cells
+                x (tuple)    - Updated crossing point (long,lat)
+            
+            Returns
+                additional_indices (list) - A list of possible cell dictionary info. None if no index added.
+                additional_cases (list) - A list of the cases connecting the additional cell indices. None if no index added.
+
+        '''
+
+        # Determine the neighbour cases if any
+        target_a_case,target_b_case = self._neighbour_case(start,end,x,case)   
+        add_indicies,add_edges = self._neighbour_cells(start,end,case,target_a_case,target_b_case)
+
+        return add_indicies,add_edges  
+
+    def diagonal_case(self,cell_a,cell_b,case):
```

Comment:
parameters `cell_a` and `cell_b` are never used; can they be removed?

Reply:
These have now been removed on the next commit